File: Best Time to Buy and Sell Stock II.cpp
// Leetcode Problem 122: Best Time to Buy and Sell Stock II
// C++ CODE
#include <vector>

class Solution {
public:
    int maxProfit(const std::vector<int>& prices) {
        int maxProfit = 0;

        // Accumulate profit from every positive daily increase
        for (size_t i = 1; i < prices.size(); ++i) {
            if (prices[i] > prices[i - 1]) {
                maxProfit += prices[i] - prices[i - 1];
            }
        }

        return maxProfit;
    }
};

File: Best Time to Buy and Sell Stock II.py
# Leetcode Problem 122: Best Time to Buy and Sell Stock II
# PYTHON CODE
class Solution:

    def maxProfit(self, prices: list[int]) -> int:
        max_profit = 0

        # Accumulate profit from every positive daily price change
        for i in range(1, len(prices)):
            if prices[i] > prices[i - 1]:
                max_profit += prices[i] - prices[i - 1]

        return max_profit